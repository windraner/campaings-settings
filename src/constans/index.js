export const MENU_LIST = [
  'Campaign details',
  'Media Collections',
  'Blacklists',
  'Media rights'
];

export const ACTIVE_TAB = 'activeTab';
export const NAME = 'name';
export const DESCRIPTION = 'description';
export const PRODUCT_CATALOGUE = 'productCatalogue';
export const IS_SHOPPABLE_LINK = 'isShoppableLink';
export const IS_MEDIA_COLLECTION = 'isMediaCollection';
export const MEDIA_COLLECTION_LIST = 'mediaCollectionList';
export const IS_PHOTO_MODERATION_QUEUE = 'isPhotosModerationQueue';
export const USER_BLACK_LIST = 'userBlackList';
export const HASHTAG_BLACK_LIST = 'hashtagBlackList';
export const ACCOUNTS_SETTINGS_ITEMS = 'accountsSettingsItems';
export const SHOW_ERROR_ALERT = 'showErrorAlert';
export const SHOW_SUCCESS_ALERT = 'showSuccessAlert';
export const IS_MEDIA_RIGHTS = 'isMediaRights';
export const HASHTAG_RIGHTS_LIST = 'hashtagRightsList';
export const TERMS_AND_CONDITIONS = 'termsAndConditions';

export const VALIDATED_FIELDS = [
  ACTIVE_TAB
];
