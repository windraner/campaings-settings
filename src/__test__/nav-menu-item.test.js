import React from 'react';
import NavMenuItem from '../components/main-content-section/nav-menu/nav-menu-item/nav-menu-item';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { ACTIVE_TAB } from '../constans';

configure({ adapter: new Adapter() });

const mock = {
  name: 'test',
  menuAlerts: [],
  activeTab: 'test',
  stateHandler() {},
};

describe('<NavMenuItem />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<NavMenuItem {...mock} />);
    expect(wrapper.find('.campaigns-settings__nav-menu-item').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__nav-menu-item').text()).toBe(mock.name);
  });
  it('Check for hover and active class', () => {
    const wrapper = shallow(<NavMenuItem {...mock} />);
    wrapper.setState({ isHovered: false });
    wrapper.setProps({ activeTab: 'notTest' });
    expect(wrapper.find('.campaigns-settings__nav-menu-item_default').length).toBe(1);
    wrapper.setState({ isHovered: true });
    expect(wrapper.find('.campaigns-settings__nav-menu-item_default').length).toBe(0);
    expect(wrapper.find('.campaigns-settings__nav-menu-item_hover').length).toBe(1);
    wrapper.setProps({ activeTab: mock.name });
    expect(wrapper.find('.campaigns-settings__nav-menu-item_hover').length).toBe(0);
    expect(wrapper.find('.campaigns-settings__nav-menu-item_active').length).toBe(1);
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<NavMenuItem {...mock} stateHandler={onClickFunc} />);
    const button = wrapper.find('.campaigns-settings__nav-menu-item');
    button.simulate('click');
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([[ACTIVE_TAB, mock.name]]);
  });
  it('Check for alert icon', () => {
    const wrapper = shallow(<NavMenuItem {...mock} />);
    expect(wrapper.find('.campaigns-settings__nav-menu-icon').length).toBe(0);
    wrapper.setProps({ menuAlerts: [''] });
    expect(wrapper.find('.campaigns-settings__nav-menu-icon').length).toBe(1);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<NavMenuItem {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.setProps({ menuAlerts: [''] });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
