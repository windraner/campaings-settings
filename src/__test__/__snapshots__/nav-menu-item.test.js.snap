// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenuItem /> matches the snapshot 1`] = `
<li
  className="campaigns-settings__nav-menu-item position-relative campaigns-settings__nav-menu-item_active"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span>
    test
  </span>
</li>
`;

exports[`<NavMenuItem /> matches the snapshot 2`] = `
<li
  className="campaigns-settings__nav-menu-item position-relative campaigns-settings__nav-menu-item_active"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span>
    test
  </span>
  <span
    className="campaigns-settings__nav-menu-icon"
  />
</li>
`;
